x-pack/libbeat/common/aws/credentials.go

```diff
@@ -52,6 +52,10 @@ type ConfigAWS struct {
 	// AssumeRoleExpiryWindow will allow the credentials to trigger refreshing prior to the credentials
 	// actually expiring. If expiry_window is less than or equal to zero, the setting is ignored.
 	AssumeRoleExpiryWindow time.Duration `config:"assume_role.expiry_window"`
+
+	// SupportsCloudConnectors indicates whether the cloud connectors flow is used.
+	// If this is true, the InitializeAWSConfig should initialize the AWS cloud connector role chaining flow.
+	SupportsCloudConnectors bool `config:"supports_cloud_connectors"`
```

Comment:
I am a bit confused RE the intention here. If `SupportsCloudConnectors` is `true`, it means that cloud connectors must be used, it's not just an optional support / capability flag. If something goes wrong during the cloud connectors init, then authentication will fail. Can we name it `UseCloudConnectors` / `UseCloudConnectorsFlow`?